src/multivariate/mvlognormal.jl

```diff
@@ -232,11 +230,26 @@ var(d::MvLogNormal) = diag(cov(d))
 entropy(d::MvLogNormal) = length(d)*(1+log2π)/2 + logdetcov(d.normal)/2 + sum(mean(d.normal))
 
 #See https://en.wikipedia.org/wiki/Log-normal_distribution
-function _rand!(rng::AbstractRNG, d::MvLogNormal, x::AbstractVecOrMat{<:Real})
-    _rand!(rng, d.normal, x)
+function rand(rng::AbstractRNG, d::MvLogNormal)
+    x = rand(rng, d.normal)
     map!(exp, x, x)
     return x
 end
+function rand(rng::AbstractRNG, d::MvLogNormal, n::Int)
+    xs = rand(rng, d.normal, n)
+    map!(exp, xs, xs)
```

Comment:
According to the docs for `map!`:
> Behavior can be unexpected when any mutated argument shares memory with any other argument.

Maybe we could broadcast here instead?
```suggestion
    xs .= exp.(xs)
```